Added get_name convenience method for Calendar

mc-borscht opened the pull request:
Requested get_name convenience method created for Calendar object.

Returns name object attribute.

Docstring indicates that a None object is returned if a name is not set - maybe this is beyond the scope of this method's knowledge though?

tobixen commented:
One would expect `Calendar.get_name()` to yield the "display name" if such a thing is set at the server side.  The `self.name` attribute is some artifact which is only set if given by the client.  Ref #128 `self.name` is going to be removed in a future release.

I don't mind having such a convenience method, but it should do the right thing - and that is to do a call on the `get_property` method.

(This comment supercedes my previous comment, so I will delete it)

mc-borscht commented:
Agreed, makes sense.

1. I've pulled the original method and replaced with a new method called **get_display_name** which follows the desired behavior.
2. Replaced all calls to get_property(dav.DisplayName()) with calls to the get_display_name method

Let me know if anything else required or if you want me to change the method name back to **get_name**

tobixen commented on caldav/objects.py:
```diff
@@ -398,7 +398,7 @@ def calendar(self, name=None, cal_id=None):
         if name and not cal_id:
             for calendar in self.calendars():
-                display_name = calendar.get_property(dav.DisplayName())
+                display_name = self.get_display_name()
```
`calendar.get_display_name()` obviously :-)

tobixen commented:
Don't worry about style test not passing.  I believe it can be auto-resolved by running `tox -e style`.  I will eventually do it at some point, but of course for annotation purposes it's better that you do it.

mc-borscht commented:
Minor formatting changes were carried out by `tox -e style` command as suggested - it passes when I re-run the command.

tobixen commented:
There is still a bug on line 401

mc-borscht commented:
Fixed the following issues and passed style test

* Bug in get_display_name method - incorrectly call of get_property instance method
* get_display_name called by each calendar variable in CalendarSet calendar method - as pointed out by your comment

Hope this closes things - sorry about that!
